frontend/controller/app/identity.js

```diff
@@ -562,5 +562,30 @@ export default (sbp('sbp/selectors/register', {
       newIEK: new Secret(newIEK),
       updateToken: new Secret(updateToken)
     })
+  },
+  'gi.app/identity/delete': async (contractID: string, wPassword: Secret<string>) => {
+    const password = wPassword?.valueOf()
+    const transientSecretKeys = []
+    let oldKeysAnchorCid
+
+    // If we're creating a new session, here we derive the IEK. This key (not
+    // the password) will be passed to the service worker.
+    if (password) {
+      try {
+        const [salt, cid] = await sbp('gi.app/identity/retrieveSalt', contractID, wPassword)
```

Comment:
What is this `cid` thing?

In the selector it's `oldContractSalt`. A salt isn't a "cid" (contractID).

1. Can you rename it from `cid` to something else so that it's not confused with "contractID"?
2. Could you please update the `'gi.app/identity/retrieveSalt'` code to very clearly comment and explain what `oldContractSalt` is and how it's different from `contractSalt`?
3. Could you add a comment here explaining why it's needed here?

Reply:
> What is this cid thing?

This is a CID / hash. https://gitlab.okturtles.org/okturtles/group-income-simple/-/wikis/E2E-Protocol/Password-Change.md

> 1. Can you rename it from cid to something else so that it's not confused with "contractID"?

Not sure, it's a CID

Reply:
Oops, I didn't notice where `'gi.app/identity/retrieveSalt'` ended and my eyes went to the end of `'gi.app/identity/updateSaltRequest'`.


OK, if it's a CID, then can you explain here in a comment why we're grabbing it and whether or not it's different from `contractID` (which we already have)? If it's different, can you explain why, and if it's not different, can you explain why we're grabbing it when we already have `contractID`?